### PR TITLE
Update examples with qdq int8 quantization config file

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Update examples with qdq int8 quantization config file
1. Clip example.
2. Vit example.
3. Bert example.
4. Resnet example.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: docs/source/examples.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
...
+||bert|[Link](https://github.com/microsoft/Olive/tree/main/examples/bert)|`CPU`: with ONNX Runtime optimizations and quantization for optimized QDQ INT8 ONNX model<br>`CPU`: with ONNX Runtime optimizations and Intel® Neural Compressor quantization for optimized INT8 ONNX model<br>`CPU`: with PyTorch QAT Customized Training Loop and ONNX Runtime optimizations for optimized ONNX INT8 model<br>`GPU`: with ONNX Runtime optimizations for CUDA EP<br>`GPU`: with ONNX Runtime optimizations for TRT EP<br>`NPU`: with ONNX Runtime optimizations for QNN EP
 ||deberta|[Link](https://github.com/microsoft/Olive/tree/main/examples/deberta)|`GPU`: Optimize Azureml Registry Model with ONNX Runtime optimizations and quantization
```
[the remark this answers is not in the document]
Comment (xiaoyu-work) [reply docs/source/examples.md:15]: I can create a new config file, but the config file will just look almost the same as the current config.

[the remark this answers is not in the document]
Comment (devang-ml) [reply docs/source/examples.md:15]: Yes, that is fine.

File: docs/source/examples.md [hunk 2]
```diff
@@ -11,7 +11,7 @@
...
-||bert|[Link](https://github.com/microsoft/Olive/tree/main/examples/bert)|`CPU`: with ONNX Runtime optimizations and quantization for optimized QDQ INT8 ONNX model<br>`CPU`: with ONNX Runtime optimizations and Intel® Neural Compressor quantization for optimized INT8 ONNX model<br>`CPU`: with PyTorch QAT Customized Training Loop and ONNX Runtime optimizations for optimized ONNX INT8 model<br>`GPU`: with ONNX Runtime optimizations for CUDA EP<br>`GPU`: with ONNX Runtime optimizations for TRT EP<br>`NPU`: with ONNX Runtime optimizations for QNN EP
+||bert|[Link](https://github.com/microsoft/Olive/tree/main/examples/bert)|`CPU`: with ONNX Runtime optimizations and quantization for optimized INT8 ONNX model<br>`CPU`: with ONNX Runtime optimizations and quantization for optimized QDQ INT8 ONNX model<br>`CPU`: with ONNX Runtime optimizations and Intel® Neural Compressor quantization for optimized INT8 ONNX model<br>`CPU`: with PyTorch QAT Customized Training Loop and ONNX Runtime optimizations for optimized ONNX INT8 model<br>`GPU`: with ONNX Runtime optimizations for CUDA EP<br>`GPU`: with ONNX Runtime optimizations for TRT EP<br>`NPU`: with ONNX Runtime optimizations for QNN EP
```
Comment (devang-ml) [inline docs/source/examples.md:14]: `CPU`: with ONNX Runtime optimizations and quantization for optimized QDQ INT8 ONNX model
<br>`CPU`: with ONNX Runtime optimizations and Intel® Neural Compressor quantization for optimized INT8 ONNX model
<br>`CPU`: with PyTorch QAT Customized Training Loop and ONNX Runtime optimizations for optimized ONNX INT8 model
<br>`GPU`: with ONNX Runtime optimizations for CUDA EP
<br>`GPU`: with ONNX Runtime optimizations for TRT EP
<br>`NPU`: with ONNX Runtime optimizations for QNN EP
<br>`QDQ`: with ONNX Runtime optimizations and int8 quantization encoded in QDQ format

See the last entry

Comment (xiaoyu-work) [reply docs/source/examples.md:14]: The first CPU example is not with QDQ. Others look good to me. Updated.

File: docs/source/examples.md [hunk 3]
```diff
@@ -20,10 +20,10 @@
...
 ||mobilenet|[Link](https://github.com/microsoft/Olive/tree/main/examples/mobilenet)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
-||clip|[Link](https://github.com/microsoft/Olive/tree/main/examples/clip)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
-||resnet|[Link](https://github.com/microsoft/Olive/tree/main/examples/resnet)|`CPU`: with ONNX Runtime static/dynamic Quantization for ONNX INT8 model<br>`CPU`: with PyTorch QAT Default Training Loop and ONNX Runtime optimizations for ONNX INT8 model<br>`CPU`: with PyTorch QAT Lightning Module and ONNX Runtime optimizations for ONNX INT8 model<br>`AMD DPU`: with AMD Vitis-AI Quantization<br>`Intel GPU`: with ONNX Runtime optimizations with multiple EPs<br>`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
+||clip|[Link](https://github.com/microsoft/Olive/tree/main/examples/clip)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP<br>`CPU`: with ONNX Runtime static QDQ Quantization for ONNX INT8 model
```
Comment (devang-ml) [inline docs/source/examples.md:23]: `QDQ`: with ONNX Runtime static QDQ Quantization for ONNX INT8 model

File: docs/source/examples.md [hunk 4]
```diff
@@ -20,10 +20,10 @@
...
+||resnet|[Link](https://github.com/microsoft/Olive/tree/main/examples/resnet)|`CPU`: with ONNX Runtime static/dynamic Quantization for ONNX INT8 model<br>`CPU`: with ONNX Runtime static Quantization for ONNX INT8 model with QDQ format<br>`CPU`: with PyTorch QAT Default Training Loop and ONNX Runtime optimizations for ONNX INT8 model<br>`CPU`: with PyTorch QAT Lightning Module and ONNX Runtime optimizations for ONNX INT8 model<br>`AMD DPU`: with AMD Vitis-AI Quantization<br>`Intel GPU`: with ONNX Runtime optimizations with multiple EPs<br>`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
```
Comment (devang-ml) [inline docs/source/examples.md:24]: same ... 'QDQ': with ONNX Runtime static ...

File: docs/source/examples.md [hunk 5]
```diff
@@ -20,10 +20,10 @@
...
 ||VGG|[Link](https://github.com/microsoft/Olive/tree/main/examples/vgg)|`Qualcomm NPU`: with SNPE toolkit
 ||inception|[Link](https://github.com/microsoft/Olive/tree/main/examples/inception)|`Qualcomm NPU`: with SNPE toolkit
 ||super resolution|[Link](https://github.com/microsoft/Olive/tree/main/examples/super_resolution)|`CPU`: with ONNX Runtime pre/post processing integration for a single ONNX model
-||Vision Transformer|[Link](https://github.com/microsoft/Olive/tree/main/examples/vit)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
+||Vision Transformer|[Link](https://github.com/microsoft/Olive/tree/main/examples/vit)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP<br>`CPU`: with ONNX Runtime static QDQ Quantization for ONNX INT8 model
```
Comment (devang-ml) [inline docs/source/examples.md:28]: same

File: examples/clip/README.md [hunk 6]
```diff
@@ -1,13 +1,48 @@
 # CLIP VIT Optimization
 This folder contains examples of CLIP VIT optimization using different workflows.
 
+- CPU: [Optimization with PTQ on CPU with QDQ format](#optimization-with-ptq-on-cpu)
```
Comment (devang-ml) [inline examples/clip/README.md:4]: QDQ: ...

I don't see an optimization here, just conversion + quantization. Let's update the README accordingly.

File: examples/clip/laion_CLIP-ViT-B-32-laion2B-s34B-b79K_ptq_qdq.json [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+                        { "name": "avg", "goal": { "type": "percent-min-improvement", "value": 0.1 } },
+                        { "name": "max" },
+                        { "name": "min" }
+                    ],
+                    "user_config": {
+                        "inference_settings": { "onnx": { "execution_provider": "CPUExecutionProvider" } }
+                    }
+                },
+                {
+                    "name": "throughput",
```
Comment (devang-ml) [inline examples/clip/laion_CLIP-ViT-B-32-laion2B-s34B-b79K_ptq_qdq.json:68]: Is the "throughput" metrics measured? It is not listed in the README along with accuracy and latency.

File: examples/clip/openai_clip-vit-base-patch16_ptq_qdq.json [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                    "type": "latency",
+                    "data_config": "metric_data_config",
+                    "sub_types": [
+                        { "name": "avg", "goal": { "type": "percent-min-improvement", "value": 0.1 } },
+                        { "name": "max" },
+                        { "name": "min" }
+                    ]
+                },
+                {
+                    "name": "throughput",
```
Comment (devang-ml) [inline examples/clip/openai_clip-vit-base-patch16_ptq_qdq.json:65]: same

File: examples/clip/openai_clip-vit-base-patch32_ptq_qdq.json [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                    "type": "latency",
+                    "data_config": "metric_data_config",
+                    "sub_types": [
+                        { "name": "avg", "goal": { "type": "percent-min-improvement", "value": 0.1 } },
+                        { "name": "max" },
+                        { "name": "min" }
+                    ]
+                },
+                {
+                    "name": "throughput",
```
Comment (devang-ml) [inline examples/clip/openai_clip-vit-base-patch32_ptq_qdq.json:65]: same

File: examples/resnet/README.md [hunk 10]
```diff
@@ -1,6 +1,7 @@
 # ResNet Optimization
 This folder contains examples of ResNet optimization using different workflows.
 - CPU: [with ONNX Runtime optimizations and static/dynamic quantization](#resnet-optimization-with-ptq-on-cpu)
+- CPU: [with ONNX Runtime optimizations and static quantization with QDQ format](#resnet-optimization-with-ptq-qdq)
```
Comment (devang-ml) [inline examples/resnet/README.md:4]: QNN: [with ONNX Runtime ...]

File: examples/vit/README.md [hunk 11]
```diff
@@ -1,13 +1,31 @@
 # Vision Transformer (ViT) Optimization
 This folder contains examples of ViT optimization using different workflows.
+- CPU: [Optimization with PTQ on CPU with QDQ format](#vit-optimization-with-cpu)
```
Comment (devang-ml) [inline examples/vit/README.md:3]: same

File: docs/source/examples.md (hunk 3 above)
Comment (xiaoyu-work) [reply docs/source/examples.md]: Done.

File: docs/source/examples.md (hunk 4 above)
Comment (xiaoyu-work) [reply docs/source/examples.md]: Done.

File: examples/clip/README.md (hunk 6 above)
Comment (xiaoyu-work) [reply examples/clip/README.md]: Done.

File: examples/clip/laion_CLIP-ViT-B-32-laion2B-s34B-b79K_ptq_qdq.json (hunk 7 above)
Comment (xiaoyu-work) [reply examples/clip/laion_CLIP-ViT-B-32-laion2B-s34B-b79K_ptq_qdq.json:68]: Updated README

File: examples/clip/openai_clip-vit-base-patch16_ptq_qdq.json (hunk 8 above)
Comment (xiaoyu-work) [reply examples/clip/openai_clip-vit-base-patch16_ptq_qdq.json:65]: Updated README

File: examples/clip/openai_clip-vit-base-patch32_ptq_qdq.json (hunk 9 above)
Comment (xiaoyu-work) [reply examples/clip/openai_clip-vit-base-patch32_ptq_qdq.json:65]: Updated README

File: examples/resnet/README.md (hunk 10 above)
Comment (xiaoyu-work) [reply examples/resnet/README.md]: Done.

File: examples/vit/README.md (hunk 11 above)
Comment (xiaoyu-work) [reply examples/vit/README.md]: Done.

File: examples/bert/bert_ptq_qdq.json [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+                    "type": "throughput",
+                    "data_config": "glue_mrpc",
+                    "sub_types": [ { "name": "avg" }, { "name": "max" }, { "name": "min" } ]
+                }
+            ]
+        }
+    },
+    "passes": {
+        "conversion": { "type": "OnnxConversion", "target_opset": 13 },
+        "transformers_optimization": { "type": "OrtTransformersOptimization" },
```
Comment (jambayk) [inline examples/bert/bert_ptq_qdq.json:51]: i don't think we want transformers optimization here. it might introduce contrib ops here. What do you think @devang-ml?

We might also require making the shapes static like for the qnn ep example.

Comment (devang-ml) [reply examples/bert/bert_ptq_qdq.json:51]: That's a good point regarding contrib ops, I missed it earlier. We want to avoid inserting contrib ops.

File: examples/resnet/resnet_ptq_qdq.json [hunk 13]
```diff
@@ -0,0 +1,72 @@
+{
+    "input_model": {
+        "type": "PyTorchModel",
+        "model_path": "models/resnet_trained_for_cifar10.pt",
+        "io_config": {
+            "input_names": [ "input" ],
+            "input_shapes": [ [ 1, 3, 32, 32 ] ],
+            "output_names": [ "output" ],
+            "dynamic_axes": { "input": { "0": "batch_size" }, "output": { "0": "batch_size" } }
```
[the remark this answers is not in the document]
Comment (devang-ml) [reply examples/resnet/resnet_ptq_qdq.json:9]: we should not have dynamic shape. The QNN example does not use it.

File: examples/bert/bert_ptq_qdq.json [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+                    "name": "throughput",
+                    "type": "throughput",
+                    "data_config": "glue_mrpc",
+                    "sub_types": [ { "name": "avg" }, { "name": "max" }, { "name": "min" } ]
+                }
+            ]
+        }
+    },
+    "passes": {
+        "conversion": { "type": "OnnxConversion", "target_opset": 13 },
```
Comment (jambayk) [inline examples/bert/bert_ptq_qdq.json:50]: static shape is missing. Can you also run this example again to confirm the accuracy of the model? The full qdq model for qnn required the attention mask surgery for good accuracy but this does not have it.

Comment (jambayk) [inline examples/bert/bert_ptq_qdq.json:50]: also maybe increase the opset to 17 or higher. so that it has the layernorm op which is only available after 17.

File: examples/test/local/test_resnet_ptq_cpu.py [hunk 15]
```diff
@@ -43,6 +44,9 @@ def setup():
 def test_resnet(sampler, execution_order, system, olive_json):
     from olive.workflows import run as olive_run
 
+    hf_token = os.environ.get("HF_TOKEN")
```
Comment (jambayk) [inline examples/test/local/test_resnet_ptq_cpu.py:47]: please don't add gated dataset and hf login to this example/test. We won't be able to run this for fork builds. Don't want gated examples for the ci pipeline.

Comment (jambayk) [reply examples/test/local/test_resnet_ptq_cpu.py:47]: maybe you could try https://huggingface.co/datasets/evanarlian/imagenet_1k_resized_256 and see if it works as a replacement.

Comment (xiaoyu-work) [reply examples/test/local/test_resnet_ptq_cpu.py]: This dataset looks good. Thanks!

**Review by jambayk — APPROVED** [review]
